tests/Unit/File/UploadFileTest.php

```diff
@@ -0,0 +1,41 @@
+<?php
+/**
+ * @copyright 2015 Contentful GmbH
+ * @license   MIT
+ */
+
+namespace Contentful\Tests\Unit\Delivery;
+
+use Contentful\File\UploadFile;
+
+class UploadFileTest extends \PHPUnit_Framework_TestCase
+{
+    /**
+     * @var UploadFile
+     */
+    protected $file;
+
+    public function setUp()
+    {
+        $this->file = new UploadFile(
+            'Nyan_cat_250px_frame.png',
+            'image/png',
+            '//images.contentful.com/cfexampleapi/4gp6taAwW4CmSgumq2ekUm/9da0cd1936871b8d72343e895a00d611/Nyan_cat_250px_frame.png'
```

Comment:
I'd test this with a non-contentful URL. Otherwise it's hard to tell the difference between this and a regular `File`/`ImageFile`. (I'd just use something with `www.example.com` as the file doesn't actually need to exist)